azurerm/internal/services/hdinsight/resourceids.go

```diff
@@ -0,0 +1,3 @@
+package hdinsight
+
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=HDInsightCluster -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.HDInsight/clusters/cluster1
```

Comment:
in general we could probably ditch the repeated package name here, e.g. `HDInsight Cluster` could become `Cluster`:

```suggestion
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=Cluster -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.HDInsight/clusters/cluster1
```

meaning that the field `ClusterName` in the generated ID Struct would become `Name` - not a big deal just a heads up :)

Reply:
I'll regenerate with this one and change it.